src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs

```diff
@@ -19,6 +19,14 @@ internal record Parameter(string Name, string? Description, CSharpType Type, Con
         public CSharpAttribute[] Attributes { get; init; } = Array.Empty<CSharpAttribute>();
         public bool IsOptionalInSignature => DefaultValue != null;
 
+        private readonly bool? _isPropertyBag;
+
+        public bool IsPropertyBag
+        {
+            get => _isPropertyBag ?? false;
+            init => _isPropertyBag = value;
```

Comment:
This record was immutable before is there a reason we want to introduce the ability for it to change after construction?

Reply:
fixed